src/Sentry.Unity/Integrations/UnityScopeIntegration.cs

```diff
@@ -0,0 +1,181 @@
+
+using System;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+using Sentry.Reflection;
+using Sentry.Integrations;
+using Sentry.Unity.Integrations;
+using UnityEngine;
+using OperatingSystem = Sentry.Protocol.OperatingSystem;
+
+namespace Sentry.Unity
+{
+    internal static class UnitySdkInfo
+    {
+        public static string Version { get; } = typeof(UnitySdkInfo).Assembly.GetNameAndVersion().Version ?? "0.0.0";
+        public const string Name = "sentry.dotnet.unity";
+        public const string PackageName = "upm:sentry.unity";
+    }
+
+    internal class UnityScopeIntegration : ISdkIntegration
+    {
+        private readonly MainThreadData _mainThreadData;
+        private readonly IApplication _application;
+
+        public UnityScopeIntegration(SentryMonoBehaviour monoBehaviour, IApplication application)
+        {
+            _mainThreadData = monoBehaviour.MainThreadData;
+            _application = application;
+        }
+
+        public void Register(IHub hub, SentryOptions options)
+        {
+            var scopeUpdater = new UnityScopeUpdater((SentryUnityOptions)options, _mainThreadData, _application);
+            hub.ConfigureScope(scopeUpdater.ConfigureScope);
```

Comment:
I like this approach of setting it up as integration and providing the callback like so.